docs/content/how-to/common/nodepool-version-compatibility.md

```
@@ -0,0 +1,37 @@
+HyperShift supports version compatibility checks between HostedCluster and NodePool. This feature ensures that NodePool versions are compatible with the HostedCluster version.
+
+## Version Compatibility Rules
+
+### Supporting Version Rules
+1. NodePool version cannot be higher than the HostedCluster version.
+2. For 4.y versions, all versions now support up to 3 minor version differences (n-3).
+   For example, a 4.18 HostedCluster supports NodePools running 4.17, 4.16, and 4.15.
+
+### Examples
+- If HostedCluster version is 4.18.5:
+    - Allowed NodePool versions: 4.18.5, 4.18.4, 4.17.z, 4.16.z, 4.15.z
```

markdownlint-cli2 (0.18.1)

Unordered list indentation
Expected: 2; Actual: 4

(MD007, ul-indent)